backend/common/services/s3_service.py

```
@@ -0,0 +1,221 @@
+                ExpiresIn=3600
+            )
+    
+    async def file_exists(self, s3_key: str) -> bool:
+        """Check if file exists in S3."""
+        try:
+            loop = asyncio.get_event_loop()
+            await loop.run_in_executor(
+                None,
+                lambda: self.s3_client.head_object(Bucket=self.bucket_name, Key=s3_key)
+            )
+            return True
```

Ruff (0.14.8)

Consider moving this statement to an `else` block

(TRY300)